presto-hive/src/main/java/io/prestosql/plugin/hive/s3/PrestoS3FileSystem.java

```diff
@@ -939,35 +1009,54 @@ private InputStream openStream(Path path, long start)
                             }
                         });
             }
-            catch (InterruptedException e) {
-                Thread.currentThread().interrupt();
-                throw new RuntimeException(e);
-            }
             catch (Exception e) {
-                throwIfInstanceOf(e, IOException.class);
-                throwIfUnchecked(e);
-                throw new RuntimeException(e);
+                throw propagate(e);
             }
         }
 
         private void closeStream()
         {
             if (in != null) {
-                try {
-                    if (in instanceof S3ObjectInputStream) {
-                        ((S3ObjectInputStream) in).abort();
-                    }
-                    else {
-                        in.close();
-                    }
-                }
-                catch (IOException | AbortedException ignored) {
-                    // thrown if the current thread is in the interrupted state
-                }
+                abortStream(in);
                 in = null;
                 STATS.connectionReleased();
             }
         }
+
+        private void checkClosed()
+                throws IOException
+        {
+            if (closed.get()) {
+                throw new EOFException(FSExceptionMessages.NEGATIVE_SEEK);
+            }
+        }
+
+        private static void abortStream(InputStream in)
+        {
+            try {
+                if (in instanceof S3ObjectInputStream) {
+                    ((S3ObjectInputStream) in).abort();
```

Comment:
`close` after `abort`? It's odd that on one branch we `abort` and in the other we `close`

Reply:
I'll change it so that the instance check is not needed.

Reply:
Actually, we don't need the `close` at all, since in our usages, it will either be a `S3ObjectInputStream` or a `ByteArrayInputStream`. But it's safer / cleaner to leave it in.